From 1135a09d43093e0d0f34bc06b5fbaf81b0eae6b6 Mon Sep 17 00:00:00 2001
From: Son Trinh <son.trinh.fz@renesas.com>
Date: Tue, 17 Mar 2026 10:56:56 +0700
Subject: [PATCH 2/9] vsp2driver: Change use of strlcpy to strscpy

From Linux kernel 6.8, strlcpy() has been removed.
Switch to strscpy() to keep the driver buildable on newer kernels.

Signed-off-by: Son Trinh <son.trinh.fz@renesas.com>
---
 vsp2driver/vsp2_drv.c   | 2 +-
 vsp2driver/vsp2_video.c | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/vsp2driver/vsp2_drv.c b/vsp2driver/vsp2_drv.c
index c8d19c3..de3ddef 100644
--- a/vsp2driver/vsp2_drv.c
+++ b/vsp2driver/vsp2_drv.c
@@ -275,7 +275,7 @@ static int vsp2_create_entities(struct vsp2_device *vsp2)
 	int ret;
 
 	mdev->dev = vsp2->dev;
-	strlcpy(mdev->model, "VSP2", sizeof(mdev->model));
+	strscpy(mdev->model, "VSP2", sizeof(mdev->model));
 	snprintf(mdev->bus_info, sizeof(mdev->bus_info), "platform:%s",
 		 dev_name(mdev->dev));
 	media_device_init(mdev);
diff --git a/vsp2driver/vsp2_video.c b/vsp2driver/vsp2_video.c
index 8e9184b..c85c76c 100644
--- a/vsp2driver/vsp2_video.c
+++ b/vsp2driver/vsp2_video.c
@@ -1,7 +1,7 @@
 /*************************************************************************/ /*
  * VSP2
  *
- * Copyright (C) 2015-2017 Renesas Electronics Corporation
+ * Copyright (C) 2015-2026 Renesas Electronics Corporation
  *
  * License        Dual MIT/GPLv2
  *
@@ -1036,8 +1036,8 @@ vsp2_video_querycap(struct file *file, void *fh, struct v4l2_capability *cap)
 			  | V4L2_CAP_VIDEO_CAPTURE_MPLANE
 			  | V4L2_CAP_VIDEO_OUTPUT_MPLANE;
 
-	strlcpy(cap->driver, "vsp2", sizeof(cap->driver));
-	strlcpy(cap->card, video->video.name, sizeof(cap->card));
+	strscpy(cap->driver, "vsp2", sizeof(cap->driver));
+	strscpy(cap->card, video->video.name, sizeof(cap->card));
 	snprintf(cap->bus_info, sizeof(cap->bus_info), "platform:%s",
 		 dev_name(video->vsp2->dev));
 
-- 
2.25.1

